`test_set_from_exportdata` failure in nightly CI

```python
     def test_set_from_exportdata(mock_volumes, globalconfig2, monkeypatch, tmp_path):
        """Test setting of index from class ExportData
    
        Args:
            mock_volumes (pd.DataFrame): a volumetric like dataframe
            globalconfig2 (dict): one global variables dict
        """
        monkeypatch.chdir(tmp_path)
        index = ["OTHER"]
        exd = ExportData(
            config=globalconfig2, table_index=index, content="timeseries", name="baretull"
        )
        path = exd.export(mock_volumes)
>       assert_correct_table_index(path, index)
```
```sh
/var/local/github/actions-runner02/_work/_temp/fmudataio-test-root/tests/test_units/test_table.py:52: in assert_correct_table_index
    meta = dict_input if isinstance(dict_input, dict) else _read_dict(dict_input)
/var/local/github/actions-runner02/_work/_temp/fmudataio-test-root/tests/test_units/test_table.py:27: in _read_dict
    path.unlink()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = PosixPath('/tmp/pytest-of-x/pytest-537/share/results/tables/baretull.csv')
missing_ok = False

    def unlink(self, missing_ok=False):
        """
        Remove this file or link.
        If the path is a directory, use rmdir() instead.
        """
        if self._closed:
            self._raise_closed()
        try:
>           self._accessor.unlink(self)
E           FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pytest-of-x/pytest-537/share/results/tables/baretull.csv'

/../lib/python3.8/pathlib.py:1324: FileNotFoundError
```
```sh
FAILED tests/test_units/test_table.py::test_set_from_exportdata - FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pytest-of-x/pytest-537/share/results/tables/baretull.csv'
```

This appears to be a new failure.

--- a/tests/test_units/test_table.py
+++ b/tests/test_units/test_table.py
@@ -14,7 +14,7 @@ from fmu.dataio.providers.objectdata._provider import objectdata_provider_factor
 from fmu.dataio.providers.objectdata._tables import _derive_index
 
 
-def _read_dict(file_path: str) -> None:
+def _read_dict(file_path: str) -> dict:
     """Reads text file into dictionary
     Args:
         file_path (string): path to generated file
@@ -23,10 +23,7 @@ def _read_dict(file_path: str) -> None:
     """
     path = Path(file_path)
     meta_path = path.parent / f".{path.name}.yml"
-    meta = yaml_load(meta_path)
-    path.unlink()
-    meta_path.unlink()
-    return meta
+    return yaml_load(meta_path)
 
 
 def assert_list_and_answer(index, answer, field_to_check):

